### ISSUE TITLE
Calculating position error: different than paper

### ISSUE THREAD

**Issue by herobd** [eriklindernoren/PyTorch-YOLOv3#83]
This might be deliberate, but I noticed that the error for box position is calculated differently than the paper. The error is calculated with the sigmoid activated output and actual offset rather than the raw output and un-sigmoided offset.
I have no idea if this actually makes a big difference.

**Comment by herobd** [comment]
See the 2.1 in the v3 paper (https://pjreddie.com/media/files/papers/YOLOv3.pdf).
I'm not positive, but I think the darknet repo implements it on the sigmoid as well. Maybe the paper's wrong.

[Darknet box error computation](https://github.com/pjreddie/darknet/blob/f6d861736038da22c9eb0739dca84003c5a5e275/src/yolo_layer.c#L98) does not undo sigmoid.

I think the sigmoid set [here](https://github.com/pjreddie/darknet/blob/master/src/yolo_layer.c#L141) in darknet, and it's unchanged when passed to delta_yolo_box() on line 190.

**Comment by glenn-jocher** [comment]
Ok. Better to follow the repo than the paper I think. In comparison to the x and y errors, the width and height errors are **not** sigmoided during training. I've always wondered about this, as this makes w and h errors larger than x and y, and creates instabilities early on (hence why the burn-in is needed for SGD training).

**Comment by Ray-Luo** [comment]
I think adding sigmoid is to impose the value to fall into [0,1] range as we are predicting x, y relative to grid box, which has uniform size.
